contracts/modules/capital/SwapOperator.sol

```diff
@@ -86,129 +78,271 @@ contract SwapOperator {
     bytes32 hash = GPv2Order.hash(order, domainSeparator);
     return hash;
   }
 
-  /**
-   * @dev Compute the UID of an order using CoW protocol's logic
-   * @param order The order
-   * @return The UID (56 bytes)
-   */
+  /// @dev Compute the UID of an order using CoW protocol's logic
+  /// @param order The order
+  /// @return The UID (56 bytes)
   function getUID(GPv2Order.Data calldata order) public view returns (bytes memory) {
     bytes memory uid = new bytes(56);
     bytes32 digest = getDigest(order);
     GPv2Order.packOrderUidParams(uid, digest, order.receiver, order.validTo);
     return uid;
   }
+  
+  /// @dev Using oracle prices, returns the equivalent amount in `toAsset` for a given `fromAssetAmount` in `fromAsset`
+  /// Supports conversions for ETH to Asset, Asset to ETH, and Asset to Asset
+  function getOracleAmount(address toAsset, address fromAsset, uint fromAssetAmount) internal view returns (uint) {
+    IPriceFeedOracle priceFeedOracle = _pool().priceFeedOracle();
+
+    if (fromAsset == address(weth)) {
+      // ETH -> toAsset
+      return priceFeedOracle.getAssetForEth(toAsset, fromAssetAmount);
+    }
+    if (toAsset == address(weth)) {
+      // fromAsset -> ETH
+      return priceFeedOracle.getEthForAsset(fromAsset, fromAssetAmount);
+    }
+    // fromAsset -> toAsset via ETH
+    uint fromAssetInEth = priceFeedOracle.getEthForAsset(fromAsset, fromAssetAmount);
+    return priceFeedOracle.getAssetForEth(toAsset, fromAssetInEth);
+  }
 
-  /**
-   * @dev Approve a given order to be executed, by presigning it on CoW protocol's settlement contract
-   * Only one order can be open at the same time, and one of the swapped assets must be ether
-   * @param order The order
-   * @param orderUID The order UID, for verification purposes
-   */
-  function placeOrder(GPv2Order.Data calldata order, bytes calldata orderUID) public onlyController {
-    // Validate there's no current order going on
-    require(!orderInProgress(), "SwapOp: an order is already in place");
+  /// @dev Validates the amount against the oracle price adjusted with max slippage tolerances
+  function verifySlippage(uint amount, uint oracleAmount, uint16 maxSlippageRatio) internal pure {
+    // Calculate slippage and minimum amount we should accept
+    uint maxSlippageAmount = (oracleAmount * maxSlippageRatio) / MAX_SLIPPAGE_DENOMINATOR;
+    uint minBuyAmountOnMaxSlippage = oracleAmount - maxSlippageAmount;
+    if (amount < minBuyAmountOnMaxSlippage) {
+      revert AmountTooLow(amount, minBuyAmountOnMaxSlippage);
+    }
+  }
 
-    // Order UID verification
-    validateUID(order, orderUID);
+  /// @dev Validates order amounts against oracle prices and slippage limits.
+  /// Uses the higher maxSlippageRatio from sell/buySwapDetails, then checks if the swap amount meets the minimum after slippage.
+  function validateOrderAmount(SwapOperation memory swapOp) internal view {
```

Comment:
Simplify the checks as discussed and you'll be able to inline `verifySlippage` as well